portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -34,51 +40,67 @@
   public class Comment {
     private String time, user, content;
 
-    public Comment(String user, String content) {
-      Date date = new Date();
-      SimpleDateFormat formatter = new SimpleDateFormat("MMM d, HH:MM a");
+    public Comment(String user, String content, String time) {
       this.user = user;
       this.content = content;
-      this.time = formatter.format(date);
+      this.time = time;
     }
   }
 
-  private List<Comment> text;
-
-  @Override
-  public void init() {
-    text = new ArrayList<>();
-  }
+  private int numCommentsDisplayed = 10;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String my_json = convertToJson(text);
+    int count = 0;
+    Query myQuery = new Query("Task").addSort("timeStamp", SortDirection.DESCENDING);
+    DatastoreService dataStore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = dataStore.prepare(myQuery);
+    List<Comment> commentList = new ArrayList<>();
+
+    for (Entity entity : results.asIterable()) {
+      String user = (String) entity.getProperty("user");
+      String content = (String) entity.getProperty("content");
+      String displayTime = (String) entity.getProperty("displayTime");
+
+      Comment myComment = new Comment(user, content, displayTime);
+      commentList.add(myComment);
```

Comment:
Instead of creating a counter variable, you can break when the length of your list is greater than 10.